types/block.go

```diff
@@ -177,6 +178,7 @@ type Header struct {
 	Height  int64     `json:"height"`
 	Time    time.Time `json:"time"`
 	NumTxs  int64     `json:"num_txs"`
+	Nonce   string    `json:"nonce"`
```

Comment:
oh hi! what's this for ?

Reply:
With the new monotonic time increment, it becomes nearly impossible to differentiate between different blocks by different proposers at different rounds... but we need to differentiate in our tests, or was it for debugging I forget.  Either way it's cheap and it ensures uniqueness of proposals regardless of what info we decide to keep in the block (e.g. proposer in the header) so lets keep this in.